compiler/rustc_expand/src/mbe/diagnostics.rs

```diff
@@ -66,7 +66,9 @@ pub(super) fn failed_to_match_macro<'cx>(
         && (matches!(expected_token.kind, TokenKind::Interpolated(_))
             || matches!(token.kind, TokenKind::Interpolated(_)))
     {
-        err.note("captured metavariables except for `$tt`, `$ident` and `$lifetime` cannot be compared to other tokens");
+        err.note("captured metavariables except for `:tt`, `:ident` and `:lifetime` cannot be compared to other tokens");
+        err.note("see https://doc.rust-lang.org/nightly/reference/macros-by-example.html#forwarding-a-matched-fragment for more information");
+        err.help("try using `:tt` instead in the macro definition");
```

Comment:
It would be good to suggest this only for macros defined locally - otherwise the suggestion isn't accurate. See lines 53 for how we'd go about checking this.